### ISSUE TITLE
DictWriter signed as unsubscriptable

### ISSUE THREAD

**Issue by clo-vis** [PyCQA/pylint#6519]
### Bug description

```python
from csv import DictWriter
from typing import IO


def f_is_for_fun(file: IO[str]) -> DictWriter[str]:
    return DictWriter(file, ["id", "name"])
```

### Command used

```shell
pylint playground.py
```

### Pylint output

```shell
************* Module playground
...
playground.py:5:35: E1136: Value 'DictWriter' is unsubscriptable (unsubscriptable-object)
```

### Expected behavior

No error, as DictWriter is defined as class DictWriter(Generic[_T]) (https://github.com/python/typeshed/blob/master/stdlib/csv.pyi)

On a side note, mypy doesn't complain about the code, but it does complain for

```
def f_is_for_fun(file: IO[str]) -> DictWriter:
    return DictWriter(file, ["id", "name"])
```

> Missing type parameters for generic type "DictWriter" [type-arg]

### Pylint version

```shell
pylint 2.13.8
astroid 2.11.4
Python 3.9.12 (tags/v3.9.12:b28265d, Mar 23 2022, 23:52:46) [MSC v.1929 64 bit (AMD64)]
```

### OS / Environment

Microsoft Windows [Version 10.0.19042.1645]

**Comment by cdce8p** [comment]
The error is correct. At runtime, `DictWriter` is unsubscriptable and will raise a `TypeError`.
You need to use string annotations or PEP585 `from __future__ import annotations`. That will also hide the pylint error.

https://mypy.readthedocs.io/en/stable/runtime_troubles.html#using-classes-that-are-generic-in-stubs-but-not-at-runtime
